### PR TITLE
rpc: avoid use of cgo by hard-coding maxPathSize

### PR DESCRIPTION
**Pull request by alienc0der**
Implemented https://github.com/ethereum/go-ethereum/pull/27447

### CODE REVIEW

**Comment by sumamu** [comment]
Has anyone resynced from scratch using this PR?

File: rpc/server/ipc_unix.go [hunk 1]
```diff
@@ -28,10 +29,17 @@ import (
...
+	// see http://man7.org/linux/man-pages/man7/unix.7.html
+	maxPathSize = int(108)
+)
+
 // ipcListen will create a Unix socket on the given endpoint.
 func ipcListen(endpoint string) (net.Listener, error) {
-	if len(endpoint) > int(max_path_size) {
-		log.Warn(fmt.Sprintf("The ipc endpoint is longer than %d characters. ", max_path_size),
+	// account for null-terminator too
+	if len(endpoint)+1 > maxPathSize {
```
Comment (sumamu) [inline rpc/server/ipc_unix.go:41]: Why did you replace it with len(endpoint)+1? The value of maxPathSize = int(108) stayed the same. Also why int(108) and not just 108?

Comment (alienc0der) [reply rpc/server/ipc_unix.go:41]: Check the comment above this line

**Review by sumamu — APPROVED** [review]

File: rpc/server/ipc_unix.go [hunk 2]
```diff
@@ -28,10 +29,17 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+const (
+	// On Linux, sun_path is 108 bytes in size
+	// see http://man7.org/linux/man-pages/man7/unix.7.html
+	maxPathSize = int(108)
```
[the remark this answers is not in the document]
Comment (0x3639) [reply rpc/server/ipc_unix.go:35]: @sumamu did you resolve this question?

**Comment by 0x3639** [comment]
These changes are consistent with this discussion here: https://github.com/ethereum/go-ethereum/pull/27447/files

**Review by 0x3639 — APPROVED** [review]

**Pull request merged**